android/build.gradle

```diff
@@ -42,6 +42,9 @@ android {
         sourceCompatibility JavaVersion.VERSION_17
         targetCompatibility JavaVersion.VERSION_17
     }
+    kotlinOptions {
+       jvmTarget = "17"
```

Comment:
I see we were already using 17 from the `targetCompatibility`, but I think that might be too high? According to [this table](https://developer.android.com/build/jdks#compileSdk), Java 17 is only fully supported starting at Android 14.

I can never figure out where this might break, but my assumption is that if we accidentally use any newer API, it might compile to something incompatible on lower Android versions.

Reply:
That is a valid concern... I believe it might have been like this when the plugin was created and we forgot to downgrade it... 

I haven't heard of any related issues, but still might be worth to downgrade it to be safe. I will do it as part of this PR.

Reply:
Downgraded and did some testing, seems to be working correctly. Lmk if you think this could have any unintended consequences @JayShortway!

Reply:
Nice, I think this is safer! I don't think there should be unintended consequences. (Famous last words 😅)  Thanks for being thorough with testing! 🙏